modulo2/projeto-whatslab/pj-whatslab/src/App.js

```diff
@@ -0,0 +1,72 @@
+import logo from './logo.svg';
+import './App.css';
+import { render } from '@testing-library/react';
+import React from "react";
+
+class App extends React.Component {
+  state = {
+    conversa: [
+      {nome: "Carol",
+       mensagem: "Oi! Está dando certo?"},
+       {nome: "Computador",
+      mensagem:"Não sei... testa ai!"}
+    ],
+
+    inputUsuario:"",
+    inputMensagem: ""
+  };
+
+novoNomeUsuario=(event)=>{
+  this.setState({inputUsuario: event.target.value})
+}
+novaMensagem=(event)=>{
+  this.setState({inputMensagem: event.target.value})
+}
+
+enviaMensagem=()=>{
+  const novaMensagem={
+    nome: this.state.inputUsuario,
+    mensagem: this.state.inputMensagem
+  };
+  const todaMensagem = {...this.state.conversa, novaMensagem};
```

Comment:
O erro aqui foi o uso de chaves em vez de colchetes! A lista de mensagens é um array!
Mudando isso o app funcionou corretamente! Foi por pouco, Parabéns!

```suggestion
  const todaMensagem = [...this.state.conversa, novaMensagem];
```